core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/SplitBalanceHelper.java

```diff
@@ -304,11 +304,13 @@ protected void zkCleanup() {
           List<String> trafTables = ZKUtil.listChildrenNoWatch(zkw, zSplitBalPathNoSlash);
           List<String> hbaseTables = ZKUtil.listChildrenNoWatch(zkw, SplitBalanceHelper.zkTable);
           
-          for(String tableName : trafTables) {
+          if(trafTables != null && hbaseTables != null) {

```

Comment:
There is a possibility that trafTables is not null but hbaseTables is null, or vice versa. In this case, the cleanup will leave something.
I am not quite clear, but I think this method is to clean up every leftover znodes. So the '&&' may not be proper here?